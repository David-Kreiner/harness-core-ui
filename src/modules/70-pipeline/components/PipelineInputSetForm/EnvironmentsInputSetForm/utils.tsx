/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { get, isEmpty } from 'lodash-es'

import { RUNTIME_INPUT_VALUE, SelectOption } from '@harness/uicore'

import { DeploymentStageConfig } from 'services/cd-ng'
import { StageElementWrapperConfig } from 'services/pipeline-ng'

import { isValueRuntimeInput } from '@common/utils/utils'

import { StageType } from '@pipeline/utils/stageHelpers'
import { DeploymentStageElementConfig } from '@pipeline/utils/pipelineTypes'

export interface PropagateSelectOption extends SelectOption {
  infraLabel?: string
}

export function getPropagateStageOptions(
  stages: StageElementWrapperConfig[],
  stageIndex: number,
  templateTypes: {
    [key: string]: string
  }
): PropagateSelectOption[] {
  if (stages.length && stageIndex > 0) {
    const getStagesAllowedforPropagate = (stageItem: StageElementWrapperConfig): boolean => {
      if (stageItem.stage?.template) {
        const isAlreadyPropagatingValue = !!(stageItem?.stage?.template?.templateInputs?.spec as DeploymentStageConfig)
          ?.environment?.useFromStage?.stage
        const stageType = get(templateTypes, stageItem.stage.template.templateRef)
        return stageType === StageType.DEPLOY && !isAlreadyPropagatingValue
      } else {
        const isAlreadyPropagatingValue = !!(stageItem?.stage?.spec as DeploymentStageConfig)?.environment?.useFromStage
          ?.stage
        const stageType = stageItem?.stage?.type
        const isSingleEnvEmpty = isEmpty(
          (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment?.environmentRef
        )
        return stageType === StageType.DEPLOY && !isSingleEnvEmpty && !isAlreadyPropagatingValue
      }
    }

    const stageWithEnvV2 = stages.slice(0, stageIndex).filter(getStagesAllowedforPropagate)
    return stageWithEnvV2.map(stageItem => {
      if (stageItem.stage?.template) {
        return {
          label: `Stage [${stageItem.stage?.name}] - Template [${stageItem.stage.template.templateRef}]`,
          value: stageItem.stage?.identifier as string,
          infraLabel: `Stage [${stageItem.stage?.name}] - Template [${stageItem.stage.template.templateRef}]`
        }
      } else {
        const singleEnvironmentRef = (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment
          ?.environmentRef
        const environmentLabel = `Environment [${singleEnvironmentRef}]`

        const singleInfrastructureRef = isValueRuntimeInput(
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment?.infrastructureDefinitions as any
        )
          ? RUNTIME_INPUT_VALUE
          : (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment?.infrastructureDefinitions?.[0]
              ?.identifier
        const infrastructureLabel = `Infrastructure [${singleInfrastructureRef}]`

        return {
          label: `Stage [${stageItem.stage?.name}] - ${environmentLabel}`,
          value: stageItem.stage?.identifier as string,
          infraLabel: `Stage [${stageItem.stage?.name}] - ${infrastructureLabel}`
        }
      }
    })
  }
  return []
}
