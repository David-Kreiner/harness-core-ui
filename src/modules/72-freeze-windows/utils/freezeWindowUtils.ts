/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FreezeDetailedResponse } from 'services/cd-ng'

export const RECURRENCE = ['Daily', 'Weekly', 'Monthly', 'Annually'] as const
export const DOES_NOT_REPEAT = 'Does not repeat'

type Scope = Exclude<FreezeDetailedResponse['freezeScope'], 'unknown' | undefined>

export const scopeText: Record<Scope, string> = {
  account: 'Account',
  org: 'Organization',
  project: 'Project'
}
