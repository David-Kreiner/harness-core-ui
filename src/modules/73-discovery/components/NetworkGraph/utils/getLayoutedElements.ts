/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import ELK, { ElkExtendedEdge, ElkNode } from 'elkjs/lib/elk.bundled'
import type { Edge } from 'reactflow'
import type { LayoutedGraph } from '../types'
import { NodeTypes } from '../constants'

const elk = new ELK()

const options = {
  'elk.algorithm': 'stress'
}

export async function getLayoutedElements({ nodes, edges }: LayoutedGraph): Promise<LayoutedGraph> {
  // Generate elk graph node from reactflow node structure
  // Get the group nodes
  const groupNodes = nodes.filter(node => node.type === NodeTypes.Group)

  // Create parent elk nodes with it's children and edges
  const parentNodes = groupNodes.map(groupNode => {
    return {
      ...groupNode,
      children: nodes
        .filter(node => node.parentNode === groupNode.id)
        .map(node => ({
          ...node,
          // Adjust the target and source handle positions based on the layout
          // direction.
          targetPosition: 'top',
          sourcePosition: 'bottom',

          // Hardcode a width and height for elk to use when layouting.
          width: 100,
          height: 100
        })),
      edges: edges.filter(
        edge => /* istanbul ignore next */ edge.data?.parentNode === groupNode.id
      ) as unknown as ElkExtendedEdge[]
    }
  })

  const graph: ElkNode = {
    id: 'root',
    layoutOptions: options,
    children: parentNodes.map(node => ({
      ...node,
      // Hardcode a width and height for elk to use when layouting.
      width: 0,
      height: 0
    })),
    edges: edges as unknown as ElkExtendedEdge[]
  }

  const elkGraph = await elk.layout(graph)

  /* istanbul ignore next */
  const flattenedNodes = elkGraph?.children?.map(node => ({
    ...node,
    // React Flow expects a position property on the node instead of `x`
    // and `y` fields.
    position: { x: node.x, y: node.y }
  }))

  /* istanbul ignore next */
  elkGraph?.children?.map(node => {
    if (node.children) {
      /* istanbul ignore next */
      flattenedNodes?.push(
        ...node.children.map(n => ({
          ...n,
          // React Flow expects a position property on the node instead of `x`
          // and `y` fields.
          position: { x: n.x, y: n.y }
        }))
      )
    }
  })

  const layoutedGraph = {
    nodes: flattenedNodes,
    edges: elkGraph.edges as unknown as Edge<unknown>[]
  } as LayoutedGraph

  return layoutedGraph
}
