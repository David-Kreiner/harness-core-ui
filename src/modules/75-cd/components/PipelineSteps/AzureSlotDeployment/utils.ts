import { defaultTo } from 'lodash-es'
import { RUNTIME_INPUT_VALUE, MultiTypeInputType } from '@harness/uicore'
import type { StageElementWrapper } from '@pipeline/utils/pipelineTypes'
import type { StageElementConfig, StageElementWrapperConfig, PipelineInfoConfig } from 'services/pipeline-ng'
import type { InfraStructureDefinitionYaml, DeploymentStageConfig, EnvironmentYamlV2 } from 'services/cd-ng'

interface EnvironmentYamlV2Runtime extends Omit<EnvironmentYamlV2, 'infrastructureDefinitions'> {
  infrastructureDefinitions?: InfraStructureDefinitionYaml[] | string
}

interface DeploymentStageConfigAzureWebName extends Omit<DeploymentStageConfig, 'environment'> {
  environment: EnvironmentYamlV2Runtime
}

export interface DeploymentStageElementConfigAzureWebName extends Omit<StageElementConfig, 'spec'> {
  spec?: DeploymentStageConfigAzureWebName
}
export type SelectedStageType = StageElementWrapper<DeploymentStageElementConfigAzureWebName>

export function getStage(stageId: string, pipeline?: PipelineInfoConfig): StageElementWrapperConfig | undefined {
  if (pipeline?.stages) {
    let responseStage: StageElementWrapperConfig | undefined = undefined
    pipeline.stages.forEach(item => {
      if (item.stage && item.stage.identifier === stageId) {
        responseStage = item
      } else if (item.parallel) {
        return item.parallel.forEach(node => {
          if (node.stage?.identifier === stageId) {
            responseStage = node
          }
        })
      }
    })
    return responseStage
  }
  return
}

export const isRuntimeEnvId = (selectedStage: SelectedStageType): boolean => {
  const envId = selectedStage?.stage?.spec?.environment?.environmentRef
  return !!(envId && envId === RUNTIME_INPUT_VALUE)
}

export const isRuntimeInfraId = (selectedStage: SelectedStageType): boolean => {
  const infraId = selectedStage?.stage?.spec?.environment?.infrastructureDefinitions
  return infraId === RUNTIME_INPUT_VALUE
}

export const isMultiEnv = (selectedStage: SelectedStageType): boolean => {
  const isEnvs = !!selectedStage?.stage?.spec?.environments || !!selectedStage?.stage?.spec?.environmentGroup
  return !!isEnvs
}

export const getEnvId = (selectedStage: SelectedStageType): string => {
  if (isMultiEnv(selectedStage || isRuntimeEnvId(selectedStage))) {
    return ''
  }

  return defaultTo(selectedStage?.stage?.spec?.environment?.environmentRef, '')
}

export const getInfraId = (selectedStage: SelectedStageType): string => {
  const envId = getEnvId(selectedStage)
  if (!envId || isMultiEnv(selectedStage)) {
    return ''
  }

  if (!isRuntimeEnvId(selectedStage) && !isRuntimeInfraId(selectedStage)) {
    const [infra] = selectedStage?.stage?.spec?.environment?.infrastructureDefinitions as InfraStructureDefinitionYaml[]
    return infra.identifier
  }

  return ''
}
/* eslint-disable */
export const getEnvIdRuntime = (stageId: string, values: any): string => {
  const stage = getStage(stageId, values) as SelectedStageType
  const envId = stage?.stage?.spec?.environment?.environmentRef
  return envId || ''
}
/* eslint-disable */
export const getInfraIdRuntime = (stageId: string, values: any): string => {
  const stage = getStage(stageId, values) as SelectedStageType

  if (stage?.stage?.spec?.environment?.infrastructureDefinitions?.[0]) {
    const [{ identifier }] = stage?.stage?.spec?.environment
      ?.infrastructureDefinitions as InfraStructureDefinitionYaml[]
    return identifier
  }
  return ''
}

export const getAllowableTypes = (selectedStage: SelectedStageType): MultiTypeInputType[] => {
  if (isMultiEnv(selectedStage)) {
    return [MultiTypeInputType.EXPRESSION]
  }

  return [MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME, MultiTypeInputType.EXPRESSION]
}
