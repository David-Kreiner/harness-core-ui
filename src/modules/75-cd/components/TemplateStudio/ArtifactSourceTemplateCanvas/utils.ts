import type { ArtifactType } from '@pipeline/components/ArtifactsSelection/ArtifactInterface'
import { ArtifactToConnectorMap } from '@pipeline/components/ArtifactsSelection/ArtifactHelper'

export interface Params {
  oldConnectorRef?: string
  newConnectorRef?: string
  artifactType: ArtifactType
}

export const isConnectorRefUpdated = (params: Params) => {
  const { oldConnectorRef, newConnectorRef, artifactType } = params
  const isConnectorRefApplicable = Boolean(ArtifactToConnectorMap[artifactType])

  return Boolean(isConnectorRefApplicable && oldConnectorRef && newConnectorRef && oldConnectorRef !== newConnectorRef)
}
