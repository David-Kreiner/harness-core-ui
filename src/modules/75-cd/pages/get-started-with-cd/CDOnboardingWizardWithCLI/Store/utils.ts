export const parseJSON = (str: string): object => {
  try {
    return JSON.parse(str)
  } catch (e) {
    return {}
  }
}
