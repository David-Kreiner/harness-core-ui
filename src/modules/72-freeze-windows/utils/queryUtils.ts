import type { SelectOption } from '@harness/uicore'
import { queryParamDecodeAll } from '@common/hooks/useQueryParams'
import type { FreezeListUrlQueryParams } from '@freeze-windows/types'
import { DEFAULT_PAGE_INDEX, DEFAULT_PAGE_SIZE, DEFAULT_PIPELINE_LIST_TABLE_SORT } from '@pipeline/utils/constants'

export const getQueryParamOptions = () => ({
  parseArrays: true,
  decoder: queryParamDecodeAll(),
  processQueryParams(params: FreezeListUrlQueryParams): FreezeListUrlQueryParams {
    return {
      ...params,
      page: params.page ?? DEFAULT_PAGE_INDEX,
      size: params.size ?? DEFAULT_PAGE_SIZE,
      sort: params.sort ?? DEFAULT_PIPELINE_LIST_TABLE_SORT
    }
  }
})

export const getDefaultCalenderFilter = (): [Date, Date] => {
  const start = new Date()
  start.setDate(start.getDate() - 7)
  start.setHours(0, 0, 0, 0)

  const end = new Date()
  end.setHours(23, 59, 59, 999)

  return [start, end]
}

export const STATUS_OPTIONS: SelectOption[] = [
  { label: 'Enabled', value: 'Enabled' },
  { label: 'Disabled', value: 'Disabled' }
]
