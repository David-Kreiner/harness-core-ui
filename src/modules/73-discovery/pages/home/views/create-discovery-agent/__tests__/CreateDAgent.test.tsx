/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState } from 'react'
import { act, fireEvent, render, waitFor } from '@testing-library/react'
import { Drawer, Position } from '@blueprintjs/core'
import { Button } from '@harness/uicore'
import * as servicediscovery from 'services/servicediscovery'
import { TestWrapper } from '@common/utils/testUtils'
import { useCreateAgent } from 'services/servicediscovery'
import routes from '@common/RouteDefinitions'
import { accountPathProps, projectPathProps, modulePathProps } from '@common/utils/routeUtils'
import CreateDAgent from '../CreateDAgent'

const handleSubmit = jest.fn()
const infraMutate = jest.fn()

jest.useFakeTimers({ advanceTimers: true })

jest.mock('@common/components/SchedulePanel/SchedulePanel', () => ({
  ...jest.requireActual('@common/components/SchedulePanel/SchedulePanel'),
  __esModule: true,
  default: () => {
    return <div className={'schedule-panel'}>Schedule Panel</div>
  }
}))

const defaultProps = {
  setDrawerOpen: () => void 0
}

const PATH = routes.toDiscovery({ ...accountPathProps, ...projectPathProps, ...modulePathProps })
const PATH_PARAMS = {
  accountId: 'accountId',
  orgIdentifier: 'default',
  projectIdentifier: 'Discovery_Test',
  module: 'chaos'
}

function WrapperComponentWithDrawerState(): React.ReactElement {
  const [isOpen, setDrawerOpen] = useState<boolean>(false)

  return (
    <div>
      <Button text={'create'} minimal icon="cross" withoutBoxShadow onClick={() => setDrawerOpen(true)} />
      <p>Create Discovery Agent</p>
      <Drawer position={Position.RIGHT} isOpen={isOpen} isCloseButtonShown={true} size={'86%'}>
        <CreateDAgent {...defaultProps} />
      </Drawer>
    </div>
  )
}

function WrapperComponent(): React.ReactElement {
  return (
    <TestWrapper>
      <Drawer position={Position.RIGHT} isOpen={true} isCloseButtonShown={true} size={'86%'}>
        <CreateDAgent {...defaultProps} />
      </Drawer>
    </TestWrapper>
  )
}

jest.mock('services/servicediscovery', () => ({
  useCreateAgent: jest.fn().mockImplementation(() => ({ mutate: infraMutate }))
}))

describe('Define Discovery Agent Creation Step', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  test('open drawer on empty state click', async () => {
    const { getByText } = render(
      <TestWrapper>
        <WrapperComponentWithDrawerState />
      </TestWrapper>
    )
    expect(getByText('Create Discovery Agent')).toBeInTheDocument()
    const createBtn = await getByText('create')

    act(() => {
      fireEvent.click(createBtn)
    })
    await waitFor(() => expect(getByText('discovery.createNewDiscoveryAgent')).toBeInTheDocument())
  })
  test('should render dAgent creation step', async () => {
    const { container, getByTestId } = render(<WrapperComponent />)
    expect(getByTestId('input')).toBeInTheDocument()
    const identifierField = getByTestId('input').firstChild as HTMLInputElement
    expect(identifierField).toHaveTextContent('discovery.dAgentName')

    const name = document.querySelector('input[name="discoveryAgentName"]') as HTMLInputElement
    fireEvent.change(name, { target: { value: 'testTemp' } })
    expect(name).toHaveValue('testTemp')
    fireEvent.change(name, { target: { value: 'test1' } })
    expect(name).toHaveValue('test1')

    const networkTraceSwitch = document.querySelector('input[name="detectNetworkTrace"]') as HTMLInputElement
    fireEvent.change(networkTraceSwitch, { target: { value: true } })
    expect(networkTraceSwitch).toBeTruthy()

    expect(container).toBeDefined()
  })

  test('should submit form data', async () => {
    const { findByText } = render(<WrapperComponent />)
    const submitBtn = await findByText('discovery.createNewDiscoveryAgent')
    act(() => {
      fireEvent.click(submitBtn)
    })

    waitFor(() => expect(handleSubmit).toHaveBeenCalled())
    expect(useCreateAgent).toBeCalled()
  })

  test('should have the correct text content', async () => {
    const { getByText } = render(<WrapperComponent />)

    expect(getByText('common.networkMap')).toBeInTheDocument()
    expect(getByText('discovery.whatIsNetworkMap')).toBeInTheDocument()
    expect(getByText('discovery.howToCreateNetworkMap')).toBeInTheDocument()
    expect(getByText('discovery.whatIsServiceDiscovery')).toBeInTheDocument()
  })

  test('should render loading view correctly', async () => {
    jest.spyOn(servicediscovery, 'useCreateAgent').mockImplementation((): any => {
      return {
        data: undefined,
        loading: true
      }
    })

    const { container } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <CreateDAgent {...defaultProps} />
      </TestWrapper>
    )
    expect(servicediscovery.useCreateAgent).toBeCalled()

    expect(container).toMatchSnapshot()
  })

  test('should render error view correctly', async () => {
    jest.spyOn(servicediscovery, 'useCreateAgent').mockImplementation((): any => {
      return {
        data: undefined,
        loading: false,
        error: {
          message: 'some error'
        }
      }
    })

    const { findByText } = render(<WrapperComponent />)
    const submitBtn = await findByText('discovery.createNewDiscoveryAgent')
    act(() => {
      fireEvent.click(submitBtn)
    })

    waitFor(() => expect(handleSubmit).toHaveBeenCalled())
    expect(useCreateAgent).toBeCalled()
  })
})
