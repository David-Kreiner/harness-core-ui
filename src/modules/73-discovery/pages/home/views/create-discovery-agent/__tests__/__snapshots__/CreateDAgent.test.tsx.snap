// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Define Discovery Agent Creation Step should render loading view correctly 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--border StyledProps--border-bottom StyledProps--padding-large"
    style="width: 100%; height: 60px;"
  >
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-h3 StyledProps--font-weight-semi-bold"
    >
      discovery.createNewDiscoveryAgent
    </p>
    <div
      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-start"
    >
      <button
        aria-label="cancel"
        class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-tertiary"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          cancel
        </span>
      </button>
      <button
        aria-label="discovery.createDiscoveryAgent"
        class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-success Button--with-current-color Button--variation Button--variation-primary"
        type="submit"
      >
        <span
          class="bp3-button-text"
        >
          discovery.createDiscoveryAgent
        </span>
      </button>
    </div>
  </div>
  <div
    class="Layout--horizontal StyledProps--main"
    style="width: 100%; height: 100%;"
  >
    <div
      class="StyledProps--font StyledProps--main overviewContainer StyledProps--background-primaryBG StyledProps--border StyledProps--border-right"
    >
      <form
        action="#"
        class="FormikForm--main form"
      >
        <div>
          <div
            class="Layout--vertical StyledProps--main formContainer gap2 StyledProps--padding-xxlarge"
            style="width: 60%;"
          >
            <div
              class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main body"
            >
              <div>
                <div
                  class="bulletIndex"
                >
                  1
                </div>
                <div
                  class="line"
                  data-testid="showLine"
                />
              </div>
              <div
                class="content"
              >
                <div
                  class="Layout--vertical StyledProps--main"
                  style="width: 900px;"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-h5 StyledProps--font-weight-semi-bold StyledProps--margin-bottom-large"
                  >
                    discovery.selectAConnector
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--margin-top-medium StyledProps--margin-bottom-large"
                    style="width: 100%;"
                  >
                    discovery.selectAConnectorDescription
                  </p>
                  <div
                    class="StyledProps--font StyledProps--main boxContainer StyledProps--background-white StyledProps--padding-medium"
                  >
                    <div
                      class="bp3-form-group"
                      data-id="null-0"
                    >
                      <label
                        class="bp3-label"
                      >
                        <span
                          class="Tooltip--acenter"
                          data-tooltip-id="selectNetworkMapConnector"
                        >
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black StyledProps--font-small StyledProps--margin-bottom-small"
                          >
                            connectors.selectConnector
                          </p>
                        </span>
                         
                        <span
                          class="bp3-text-muted"
                        />
                      </label>
                      <div
                        class="bp3-form-content"
                      >
                        <button
                          class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--iconOnly Button--without-shadow Button--withRightIcon"
                          data-testid="cr-field-connectorRef"
                          style="width: 400px;"
                          type="button"
                        >
                          <span
                            class="bp3-button-text"
                          >
                            <span
                              class="placeholder"
                            >
                              connectors.selectConnector
                            </span>
                          </span>
                          <span
                            class="bp3-icon bp3-icon-chevron-down StyledProps--main"
                            icon="chevron-down"
                          >
                            <svg
                              data-icon="chevron-down"
                              height="14"
                              viewBox="0 0 16 16"
                              width="14"
                            >
                              <desc>
                                chevron-down
                              </desc>
                              <path
                                d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                    </div>
                    <div
                      data-testid="input"
                      style="width: 400px;"
                    >
                      <div
                        class="InputWithIdentifier--txtNameContainer TextInput--main"
                      >
                        <div
                          class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
                        >
                          <div
                            class="Text--withDocsTooltip"
                            data-tooltip-id="universalInputNameWithId"
                          >
                            <p
                              class="StyledProps--font StyledProps--main"
                            >
                              Id
                            </p>
                          </div>
                          <p
                            class="StyledProps--font StyledProps--main"
                          >
                            :
                          </p>
                          <span
                            class="bp3-popover-wrapper"
                          >
                            <span
                              class="bp3-popover-target"
                            >
                              <div
                                class="bp3-editable-text bp3-editable-text-placeholder InputWithIdentifier--idValue"
                                tabindex="0"
                              >
                                <span
                                  class="bp3-editable-text-content"
                                  style="height: 1px; line-height: 1px; min-width: 0;"
                                />
                              </div>
                            </span>
                          </span>
                          <span
                            class="bp3-icon StyledProps--main"
                            data-icon="Edit"
                            style="vertical-align: middle; cursor: pointer;"
                          >
                            <svg
                              height="12"
                              viewBox="0 0 16 16"
                              width="12"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                fill="currentColor"
                                fill-rule="evenodd"
                              />
                              <path
                                clip-rule="evenodd"
                                d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                fill="currentColor"
                                fill-rule="evenodd"
                                opacity="0.3"
                              />
                            </svg>
                          </span>
                        </div>
                        <div
                          class="bp3-form-group"
                          data-id="discoveryAgentName-1"
                        >
                          <label
                            class="bp3-label"
                            for="discoveryAgentName"
                          >
                            discovery.dAgentName
                             
                            <span
                              class="bp3-text-muted"
                            />
                          </label>
                          <div
                            class="bp3-form-content"
                          >
                            <div
                              class="bp3-input-group"
                            >
                              <input
                                autocomplete="off"
                                class="bp3-input"
                                name="discoveryAgentName"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="bp3-form-group"
                        data-id="discoveryNamespace-2"
                      >
                        <label
                          class="bp3-label"
                          for="discoveryNamespace"
                        >
                          common.namespace
                           
                          <span
                            class="bp3-text-muted"
                          />
                        </label>
                        <div
                          class="bp3-form-content"
                        >
                          <div
                            class="bp3-input-group"
                          >
                            <input
                              autocomplete="off"
                              class="bp3-input"
                              name="discoveryNamespace"
                              placeholder="discovery.discoveryNamespacePlaceholder"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main body StyledProps--margin-bottom-large"
            >
              <div>
                <div
                  class="bulletIndex"
                >
                  2
                </div>
              </div>
              <div
                class="content"
              >
                <div
                  class="Layout--vertical StyledProps--main margin2"
                  style="width: 900px;"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-h5 StyledProps--font-weight-semi-bold StyledProps--margin-bottom-large"
                  >
                    discovery.dataCollectionSettings common.optionalLabel
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--margin-top-medium StyledProps--margin-bottom-large"
                    style="width: 100%;"
                  >
                    discovery.dataCollectionSettingsDesc
                  </p>
                  <div
                    class="StyledProps--font StyledProps--main boxContainer StyledProps--background-white StyledProps--padding-medium"
                  >
                    <label
                      class="bp3-control bp3-switch StyledProps--font Switch--switch StyledProps--main StyledProps--font-weight-semi-bold StyledProps--font-size-normal StyledProps--color StyledProps--color-grey900 StyledProps--margin-bottom-medium"
                    >
                      <input
                        name="detectNetworkTrace"
                        type="checkbox"
                      />
                      <span
                        class="bp3-control-indicator"
                      />
                      discovery.detectNetworkTrace
                    </label>
                    <div
                      class="bp3-form-group"
                      data-id="blacklistedNamespaces-3"
                    >
                      <label
                        class="bp3-label"
                        for="blacklistedNamespaces"
                      >
                        discovery.blacklistedNamespaces common.optionalLabel
                         
                        <span
                          class="bp3-text-muted"
                        />
                      </label>
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="bp3-popover-wrapper TagInput--tagInput bp3-fill"
                        >
                          <div
                            class="bp3-popover-target"
                          >
                            <div
                              class=""
                            >
                              <div
                                class="bp3-input bp3-tag-input bp3-fill bp3-multi-select"
                              >
                                <div
                                  class="bp3-tag-input-values"
                                >
                                  <input
                                    class="bp3-input-ghost bp3-multi-select-tag-input-input"
                                    placeholder="Search..."
                                    value=""
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="schedule-panel"
                    >
                      Schedule Panel
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="details"
    >
      <div
        class="Layout--vertical StyledProps--main StyledProps--padding-top-xxlarge StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--padding-bottom-xxlarge"
        style="width: 100%;"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--margin-bottom-large"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-h5 StyledProps--font-weight-semi-bold"
          >
            common.networkMap
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-small-bold StyledProps--color StyledProps--color-primary7"
            style="display: flex; align-items: center;"
          >
            learnMore
            <span
              class="bp3-icon StyledProps--main Text--icon StyledProps--padding-left-xsmall StyledProps--color StyledProps--color-primary7"
              data-icon="main-share"
            >
              <svg
                height="10"
                viewBox="0 0 14 14"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.5 12.6H1.4V3.5h6.1l1.4-1.4H.7c-.4 0-.7.3-.7.7v10.5c0 .4.3.7.7.7h10.5c.4 0 .7-.3.7-.7V5.1l-1.4 1.4zM13.3 0H8.4c-.4 0-.7.3-.7.7s.3.7.7.7h3.2L6.5 6.5c-.1.1-.2.3-.2.5 0 .4.3.7.7.7.2 0 .4-.1.5-.2l5.1-5.1v3.2c0 .4.3.7.7.7s.7-.3.7-.7V.7c0-.4-.3-.7-.7-.7z"
                  fill="#4d5969"
                />
              </svg>
            </span>
          </p>
        </div>
        <img
          alt="Network Map"
          class="image"
          src="test-file-stub"
        />
        <div
          class="StyledProps--font StyledProps--main StyledProps--margin-top-medium StyledProps--margin-bottom-xlarge"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-h6 StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey900 StyledProps--margin-bottom-xlarge"
          >
            discovery.whatIsNetworkMap
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--font-weight-light StyledProps--color StyledProps--color-grey600"
          >
            discovery.networkMapDescription
          </p>
        </div>
        <div
          class="StyledProps--font StyledProps--main StyledProps--margin-top-medium StyledProps--margin-bottom-xlarge"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-h6 StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey900 StyledProps--margin-bottom-xlarge"
          >
            discovery.howToCreateNetworkMap
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--font-weight-light StyledProps--color StyledProps--color-grey600"
          >
            discovery.howToCreateNetworkMapDesc
          </p>
        </div>
        <div
          class="StyledProps--font StyledProps--main StyledProps--margin-top-medium"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-h6 StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey900 StyledProps--margin-bottom-xlarge"
          >
            discovery.whatIsServiceDiscovery
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--font-weight-light StyledProps--color StyledProps--color-grey600"
          >
            discovery.whatIsServiceDiscoveryDesc
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
